lib/raxx/stack.ex

```diff
@@ -0,0 +1,127 @@
+defmodule Raxx.Stack do
+  alias Raxx.Server
+  alias Raxx.Middleware
+  alias Raxx.Middleware.Pipeline
+
+  @behaviour Server
+
+  @enforce_keys [:pipeline, :server]
+  defstruct @enforce_keys
+
+  @moduledoc """
+  `Raxx.Stack` implements `Raxx.Server` behaviour that works with
+  `t:Raxx.Stack.t/0` as its state.
+  """
+
+  @typedoc """
+  A `t:Raxx.Stack.t/0` represents a pipeline of middlewares attached to a server.
+
+  NOTE: Don't rely on the internal structure of this type. It can be modified
+  at an arbitrary moment to improve performance or capabilities.
+  """
+  # DEBT: compare struct t() performance to a (tagged) tuple implementation
+  @opaque t :: %__MODULE__{
+            pipeline: Pipeline.t(),
```

Comment:
Could this just be middlewares (instead of pipeline)?
The type `%Raxx.Stack{middlewares: list_of_middlewares, server: server}` is nearly as informative as the same but with list_of_middlewares replaced with `pipeline_of_middlewares

Reply:
sure